File: 백준/Silver/31246. 모바일 광고 입찰/모바일 광고 입찰.py
import sys
input = sys.stdin.readline

n, k = map(int,input().split())
lst = []
for _ in range(n):
    a, b = map(int,input().split())
    lst.append(b-a)

lst.sort()
if lst[k-1] < 0:
    print(0)
else:
    print(lst[k-1])

File: 백준/Silver/31246. 모바일 광고 입찰/모바일 광고 입찰.js
const fs = require("fs");

const inputData = fs.readFileSync("/dev/stdin").toString().trim().split("\n");

const n = +inputData[0].split(" ")[0];
const k = +inputData[0].split(" ")[1];

const dif = [];

for (let i = 1; i <= n; i++) {
  const a = +inputData[i].split(" ")[0];
  const b = +inputData[i].split(" ")[1];

  dif.push(b - a);
}

dif.sort(function (a, b) {
  return a - b;
});

// console.log(dif);

if (dif[k - 1] < 0) {
  console.log(0);
} else {
  console.log(dif[k - 1]);
}
